test/membrane_rtsp/server/server_logic_test.exs

```diff
@@ -0,0 +1,234 @@
+defmodule Membrane.RTSP.ServerLogicTest do
+  @moduledoc false
+
+  use ExUnit.Case
+
+  import Mockery
+
+  alias Membrane.RTSP.{Request, Response}
+  alias Membrane.RTSP.Server.FakeHandler
+  alias Membrane.RTSP.Server.{Logic, Logic.State}
+
+  @url %URI{scheme: "rtsp", host: "localhost", port: 554, path: "/stream"}
+
+  setup_all do
+    state = %State{
+      socket: %{},
+      request_handler: FakeHandler,
+      request_handler_state: FakeHandler.handle_open_connection(nil)
+    }
+
+    [state: state]
+  end
+
+  test "handle OPTIONS request", %{state: state} do
+    mock(:gen_tcp, [send: 2], fn %{}, response ->
+      assert response =~ "200"
+      assert response =~ "\r\nPublic: #{Enum.join(Logic.allowed_methods(), ", ")}\r\n"
+    end)
+
+    assert {:ok, ^state} =
+             %Request{method: "OPTIONS"}
+             |> Request.stringify(@url)
+             |> Logic.process_request(state)
+  end
+
+  test "handle GET_PARAMETER request", %{state: state} do
+    mock(:gen_tcp, [send: 2], fn %{}, response -> assert response =~ "200" end)
+
+    assert {:ok, ^state} =
+             %Request{method: "GET_PARAMETER"}
+             |> Request.stringify(@url)
+             |> Logic.process_request(state)
+  end
+
+  test "handle DESCRIBE request", %{state: state} do
+    media = ExSDP.Media.new(:video, 0, "RTP/AVP", 96)
+    sdp = ExSDP.new() |> ExSDP.add_media(%{media | connection_data: nil})
+
+    mock(FakeHandler, [respond: 2], fn request, %{} ->
+      response =
+        Response.new(200)
+        |> Response.with_header("Content-Type", "application/sdp")
+        |> Response.with_body(to_string(sdp))
+
+      {response, %{described_url: request.path}}
+    end)
+
+    mock(:gen_tcp, [send: 2], fn %{}, response ->
+      assert response =~ "200"
```

Comment:
Can we check the response code more thoroughly? Right now it would match 200 anywhere in the response